kaldb/src/main/java/com/slack/kaldb/writer/MurronLogFormatter.java

```diff
@@ -44,6 +50,41 @@ public static Trace.Span fromApiLog(Murron.MurronMessage murronMsg)
         "");
   }
 
+  public static Trace.Span fromIngestDocument(IngestDocument ingestDocument) {
+    ZonedDateTime timestamp =
+        (ZonedDateTime)
+            ingestDocument
+                .getIngestMetadata()
+                .getOrDefault("timestamp", ZonedDateTime.now(ZoneOffset.UTC));
+    // Trace.Span expects duration in microseconds today
+    long epochMicro = timestamp.toInstant().toEpochMilli() / 1000;
```

Comment:
Microseconds should be milliseconds multiplied by 1000, not divided. We probably should add a test assert against our date handling too, as that'd be a bit of a pain to run down in the future if it regressed.